File: Backend/models/chofer.py
from sqlmodel import SQLModel, Field, Relationship
from typing import Optional, List
import datetime
from sqlalchemy import Column, DateTime

from models.bus import Bus

class ChoferBase(SQLModel):
    nombre: str
    apellido: Optional[str] = None
    dni: Optional[str] = None
    telefono: Optional[str] = None
    foto_url: Optional[str] = None
    qr_pago_url: Optional[str] = None
    licencia_conducir: Optional[str] = None

class Chofer(ChoferBase, table=True):
    IdChofer: Optional[int] = Field(default=None, primary_key=True)
    fecha_ingreso: datetime.datetime = Field(
        default_factory=datetime.datetime.utcnow,
        sa_column=Column(DateTime(timezone=True))
    )
    estado: bool = True

    buses: List["Bus"] = Relationship(back_populates="chofer")

class ChoferCreate(SQLModel):
    nombre: str
    apellido: Optional[str] = None
    dni: Optional[str] = None
    telefono: Optional[str] = None

class ChoferPublic(ChoferBase):
    IdChofer: int
    fecha_ingreso: datetime.datetime
    estado: bool

class ChoferUpdate(SQLModel):
    nombre: Optional[str] = None
    apellido: Optional[str] = None
    dni: Optional[str] = None
    telefono: Optional[str] = None

File: flutter/lib/data/models/chofer.dart
class Chofer {
  final int idChofer;
  final String nombre;
  final String? apellido;
  final String? dni;
  final String? telefono;
  final String? fotoUrl;
  final String? qrPagoUrl;
  final String? licenciaConducir;
  final DateTime fechaIngreso;
  final bool estado;

  Chofer({
    required this.idChofer,
    required this.nombre,
    this.apellido,
    this.dni,
    this.telefono,
    this.fotoUrl,
    this.qrPagoUrl,
    this.licenciaConducir,
    required this.fechaIngreso,
    required this.estado,
  });

  factory Chofer.fromJson(Map<String, dynamic> json) {
    return Chofer(
      idChofer: json['IdChofer'],
      nombre: json['nombre'],
      apellido: json['apellido'],
      dni: json['dni'],
      telefono: json['telefono'],
      fotoUrl: json['foto_url'],
      qrPagoUrl: json['qr_pago_url'],
      licenciaConducir: json['licencia_conducir'],
      fechaIngreso: DateTime.parse(json['fecha_ingreso']),
      estado: json['estado'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'nombre': nombre,
      'apellido': apellido,
      'dni': dni,
      'telefono': telefono,
      'foto_url': fotoUrl,
      'qr_pago_url': qrPagoUrl,
      'licencia_conducir': licenciaConducir,
    };
  }
}